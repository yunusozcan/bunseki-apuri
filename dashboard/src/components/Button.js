import React from 'react';
import '../stylesheets/components/Button.css';

function Button(props) {
    const {children, onClick} = props;

    return (
        <button className="App-button" type="button" onClick={onClick}>
            {children}
        </button>
    );
}

export default Button;
