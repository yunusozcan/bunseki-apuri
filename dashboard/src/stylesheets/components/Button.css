.App-button {
    flex-grow: 1;
    background: white;
    border: none;
    float: right;
}

.App-button span, .App-button img {
    vertical-align: middle;
}
