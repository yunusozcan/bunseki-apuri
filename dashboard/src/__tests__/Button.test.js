import React from 'react';
import Button from '../components/Button';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<Button onClick={()=> {console.log('test')}}>Test</Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
