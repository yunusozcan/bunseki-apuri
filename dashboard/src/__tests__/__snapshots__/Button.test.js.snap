// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="App-button"
  onClick={[Function]}
  type="button"
>
  Test
</button>
`;
